### PR TITLE
moved MessageNumberCSV to network.py and added helper functions in util.py

### PR DESCRIPTION
**Pull request by krishnacharya**
## What? Why?

Changes proposed in this pull request:
- Moves MessageNumberCSV to network.py (**main refactoring**)
-adds helper functions create_connected_nick_list and correct_last_char_list in util.py
-adds helper functions  print_node_degree and create_adj_matrix in channel_user_presence_graph_and_csv function of network.py (**minor**)
-adds helper functions nick_same_list_to_conn_comp_list and nick_receiver_from_conn_comp(**minor**) 

## Checks
- [ ] Single commit and [No merge commits](http://nathanleclaire.com/blog/2014/09/14/dont-be-scared-of-git-rebase/)
- [x] Make sure you have added required documentation
- [x] If you have refactored a code, make sure that you have compared the outputs pre and post refactoring. Add both pre and post refaoring screenshots below if the output is visual

#any output files
before:
[#kubuntu-devel_2013_1_1_output-parser-bins.xlsx](https://github.com/prasadtalasila/IRCLogParser/files/819145/kubuntu-devel_2013_1_1_output-parser-bins.xlsx)

after:
[Message_number_bins.xlsx](https://github.com/prasadtalasila/IRCLogParser/files/819147/Message_number_bins.xlsx)

(I have added column AW for reviewers convenience it gives the total number of messages sent in a day, the output is only till column AV)
Particularly notice January 12th (the 12th row in the xlsx files) earlier the total number of messages were 2, whereas the right value is 3 which the current refactoring takes care of. 

as seen in this graph:
mtg12

## Notify reviewers
@rohangoel96

### CODE REVIEW

**Comment by krishnacharya** [comment]
The code climate score may drop as i have removed an 'A' file MessageNumberCSV and moved it to network.py which requires little more refactoring.

File: lib/util.py [hunk 1]
```diff
@@ -142,3 +142,20 @@ def extend_conversation_list(nick_sender, nick_receiver, conversation):
...
     return conversation
+
+def create_connected_nick_list(conn_comp_list):
+    """ A function that converts each individual list member to a list 
+
+    """
+    for i in range(len(conn_comp_list)):
+        conn_comp_list[i] = list(conn_comp_list[i])
+
+def correct_last_char_list(rec_list):
```
Comment (rohangoel96) [inline lib/util.py:153]: https://github.com/prasadtalasila/IRCLogParser/blob/dev/lib/util.py#L90
are these related? can these be merged?

File: sample.py [hunk 2]
```diff
@@ -18,9 +18,9 @@
 # message_number_graph_day_list = network.message_number_graph(log_data, nicks, nick_same_list, DAY_BY_DAY_ANALYSIS = True)
 #degree_anal_message_numder = network.degree_analysis_on_graph(message_number_graph)
 # message_time_graph_list = network.create_message_time_graph(log_data, nicks, nick_same_list)
-out_degree_node_number, in_degree_node_number, total_degree_node_number = network.degreeNodeNumberCSV(log_data, nicks, nick_same_list)
```
Comment (rohangoel96) [inline sample.py]: i forgot this in the last PR, correct the name here to follow _ and not camel case

File: sample.py [hunk 3]
```diff
@@ -18,9 +18,9 @@
...
+#out_degree_node_number, in_degree_node_number, total_degree_node_number = network.degreeNodeNumberCSV(log_data, nicks, nick_same_list)
 #nick_change_graph_list =  user.nick_change_graph(log_data)
-
+bin_matrix, ans = network.create_message_number_binsCSV(log_data, nicks, nick_same_list)
```
Comment (rohangoel96) [inline sample.py]: ans is not a good variable name here.
also i think it'll be better if we skip the word create in these functions which the user will use.
I mean this is redundant right? obviously it will create the message number bins.
i feel `message_number_binsCSV` is better naming. what do you think?

File: sample.py (hunk 2 above)
Comment (krishnacharya) [reply sample.py]: yes, i'll do that

File: lib/config.py [hunk 4]
```diff
@@ -34,6 +34,9 @@
 GENERATE_DEGREE_ANAL = False
 PRINT_CHANNEL_USER_HASH = False
 
+#MessageTimeGraph and NickChange constants
```
Comment (rohangoel96) [inline lib/config.py]: this is not needed. we have included this in function definition.
check the function definitions if messageTime and NickChange

File: lib/config.py [hunk 5]
```diff
@@ -1,6 +1,6 @@
 DEBUGGER = True
-LOG_DIRECTORY = "/home/rohan/parser_files/YearlyLogFiles/"
-OUTPUT_DIRECTORY = "/home/rohan/Desktop/"
+LOG_DIRECTORY = "/media/krishna/Media/ParserIn/YearlyLogFiles/"
```
Comment (rohangoel96) [inline lib/config.py]: undo this

File: sample.py (hunk 3 above)
Comment (krishnacharya) [reply sample.py]: yes i feel its implied, ok will change it

File: lib/analysis/network.py [hunk 6]
```diff
@@ -45,26 +44,27 @@ def message_number_graph(log_dict, nicks, nick_same_list, DAY_BY_DAY_ANALYSIS =
                     corrected_nick = util.correctLastCharCR(parsed_nick.group(0)[1:-1])
                     nick_sender = ""
                     nick_receiver = ""
-                    for i in range(config.MAX_EXPECTED_DIFF_NICKS):
-                        if ((i < len(conn_comp_list)) and (corrected_nick in conn_comp_list[i])):  #change nick_same_list to conn_comp_list because conn_comp_list is the main list of all users and nicks now
-                            nick_sender = conn_comp_list[i][0]
-                            break
+                    nick_sender = nick_same_list_to_conn_comp_list(conn_comp_list, corrected_nick)
```
Comment (rohangoel96) [inline lib/analysis/network.py]: I'm not really sure if this function will work correctly here, since the break might not be executed.
Please confirm this.
Also, check outputs of some other graphs to confirm that you aren't breaking any code.

File: lib/analysis/network.py [hunk 7]
```diff
@@ -45,26 +44,27 @@ def message_number_graph(log_dict, nicks, nick_same_list, DAY_BY_DAY_ANALYSIS =
...
                                 if(corrected_nick != nick):
-                                    for i in range(config.MAX_EXPECTED_DIFF_NICKS):
-                                        if ((i<len(conn_comp_list)) and (nick in conn_comp_list[i])):
-                                            nick_receiver=conn_comp_list[i][0]
-                                            break
+                                    #for i in range(config.MAX_EXPECTED_DIFF_NICKS):
+                                        #if ((i<len(conn_comp_list)) and (nick in conn_comp_list[i])):
+                                            #nick_receiver=conn_comp_list[i][0]
+                                            #break
+                                    nick_receiver = nick_same_list_to_conn_comp_list(conn_comp_list, nick)
```
Comment (rohangoel96) [inline lib/analysis/network.py]: I'm not really sure if this function will work correctly here, since the break might not be executed.
Please confirm this.
Also, check outputs of some other graphs to confirm that you aren't breaking any code.

File: lib/util.py (hunk 1 above)
Comment (krishnacharya) [reply lib/util.py:153]: L90 is for rec_list_splicing whereas L153 is more linked with correctLastCharCR. So i don't think merging them do much good.

File: lib/analysis/channel.py [hunk 8]
```diff
@@ -226,9 +226,10 @@ def response_time(log_dict, nicks, nick_same_list):
 
 						if not rec_list[1]:
 							break
-						for i in range(len(rec_list)):
-							if(rec_list[i]):
-								rec_list[i] = util.correctLastCharCR(rec_list[i])
+						#for i in range(len(rec_list)):
```
Comment (rohangoel96) [inline lib/analysis/channel.py]: why have you left these as comments? delete ?

File: lib/analysis/network.py (hunk 7 above)
Comment (krishnacharya) [reply lib/analysis/network.py]: i tried this out and it works fine.The Message Time graphs work well with this.

File: sample.py (hunk 3 above)
Comment (rohangoel96) [reply sample.py]: also keep csv in small case

File: lib/util.py (hunk 1 above)
Comment (rohangoel96) [reply lib/util.py:153]: cool.

**Comment by krishnacharya** [comment]
I have added functionality for variable size bin length
for eg for a BIN_LENGTH_MINS = 20 we have : 
[MessageNumber_binsize_20.xlsx](https://github.com/prasadtalasila/IRCLogParser/files/819265/MessageNumber_binsize_20.xlsx)
and as earlier for BIN_LENGTH_MINS = 30 we have:
[Message_number_networknew30.xlsx](https://github.com/prasadtalasila/IRCLogParser/files/819267/Message_number_networknew30.xlsx)
 In general the bin length in minutes can be adjusted via the config.BIN_LENGTH_MINS 's value

**Comment by krishnacharya** [comment]
The helper functions in util.py as well as those added in network.py all give the same output as before refactoring 
Some comparisons are as follows:
**previous** message number aggregate (between 12th and 13th Jan)
message_number_graph
message number aggregate (b/w 12th and 13th Jan) **after** refactoring:
mnagg

A single days message number graph **before** (9th Jan):
mng9
**after** 
mng9

keywords log output **before**:
[keywordsold.txt](https://github.com/prasadtalasila/IRCLogParser/files/821299/keywordsold.txt)
**after**
[keywordsnew.txt](https://github.com/prasadtalasila/IRCLogParser/files/821300/keywordsnew.txt)

convl,conv_ref_time,resp_time **before**:
convl:
[conv_len.xlsx](https://github.com/prasadtalasila/IRCLogParser/files/821303/conv_len.xlsx)
conv_ref_time:
[conv_ref_time.xlsx](https://github.com/prasadtalasila/IRCLogParser/files/821305/conv_ref_time.xlsx)
resp_time:
[resp_time.xlsx](https://github.com/prasadtalasila/IRCLogParser/files/821309/resp_time.xlsx)

**after**:
convl:
[conv_len.xlsx](https://github.com/prasadtalasila/IRCLogParser/files/821314/conv_len.xlsx)
conv_ref_time:
[conv_ref_time.xlsx](https://github.com/prasadtalasila/IRCLogParser/files/821315/conv_ref_time.xlsx)
resp_time:
[resp_time.xlsx](https://github.com/prasadtalasila/IRCLogParser/files/821317/resp_time.xlsx)

Message time graph:
**prev**:
mtg13

**after**:
mtg13

After these checks i can safely say that the refactoring works fine

**Comment by rohangoel96** [comment]
awesome work there @krishnacharya
